testctx/testctx.go

```diff
@@ -1,366 +0,0 @@
-//	    t.Run("subtest", func(ctx context.Context, t *testctx.T) {
-//	        // Use context for deadlines, tracing, etc
-//	        // Use t.Fatal, t.Error, etc as normal
-//	    })
-//	}
-package testctx
-
-import (
-	"context"
-	"fmt"
-	"os"
-	"reflect"
-	"strings"
-	"sync"
-	"testing"
-	"time"
-
-	"go.opentelemetry.io/otel/attribute"
-	"go.opentelemetry.io/otel/codes"
-	"go.opentelemetry.io/otel/log"
-	"go.opentelemetry.io/otel/trace"
-)
-
-type T = TB[*testing.T]
-type B = TB[*testing.B]
-
-type MiddlewareT = Middleware[*testing.T]
-type MiddlewareB = Middleware[*testing.B]
-
-type Middleware[Type RunnableTB[Type]] func(*TB[Type]) *TB[Type]
-
-type RunnableTB[Type testing.TB] interface {
-	testing.TB
-	Run(string, func(Type)) bool
-}
-
-func Run(ctx context.Context, testingT *testing.T, suite any, middleware ...Middleware[*testing.T]) {
-	t := New(ctx, testingT)
-	for _, m := range middleware {
-		t = m(t)
-	}
-
-	suiteT := reflect.TypeOf(suite)
-	suiteV := reflect.ValueOf(suite)
-
-	for i := range suiteV.NumMethod() {
-		methT := suiteT.Method(i)
-		if !strings.HasPrefix(methT.Name, "Test") {
-			continue
-		}
-
-		methV := suiteV.Method(i)
-		// is it possible to also accept func(context.Context, *testing.T) ?
-		tf, ok := methV.Interface().(func(context.Context, *TB[*testing.T]))
-		if !ok {
-			testingT.Fatalf("suite method %s does not have the right signature; must be func(testctx.T); have %T",
-				methT.Name,
-				methV.Interface())
-		}
-
-		t.Run(methT.Name, tf)
-	}
-}
-
-func Bench(ctx context.Context, testingB *testing.B, suite any, middleware ...Middleware[*testing.B]) {
-	b := New(ctx, testingB)
-	for _, m := range middleware {
-		b = m(b)
-	}
-
-	suiteT := reflect.TypeOf(suite)
-	suiteV := reflect.ValueOf(suite)
-
-	for i := range suiteV.NumMethod() {
-		methT := suiteT.Method(i)
-		if !strings.HasPrefix(methT.Name, "Benchmark") {
-			continue
-		}
-
-		methV := suiteV.Method(i)
-		bf, ok := methV.Interface().(func(context.Context, *TB[*testing.B]))
-		if !ok {
-			testingB.Fatalf("suite method %s does not have the right signature; must be func(*testctx.TB[*testing.B]); have %T",
-				methT.Name,
-				methV.Interface())
-		}
-
-		b.Run(methT.Name, bf)
-	}
-}
-
-func New[Type RunnableTB[Type]](ctx context.Context, t Type) *TB[Type] {
-	// Interrupt the context when the test is done.
-	ctx, cancel := context.WithCancel(ctx)
-	t.Cleanup(cancel)
-	return &TB[Type]{
-		RunnableTB: t,
-		ctx:        ctx,
-		mu:         new(sync.Mutex),
-	}
-}
-
-func WithParallel(t *TB[*testing.T]) *TB[*testing.T] {
-	t.RunnableTB.(*testing.T).Parallel()
-	return t.
-		BeforeEach(func(t *TB[*testing.T]) *TB[*testing.T] {
-			t.RunnableTB.(*testing.T).Parallel()
-			return t
-		})
-}
-
-func N(b *TB[*testing.B]) int {
-	return b.RunnableTB.(*testing.B).N
-}
-
-const TestCtxTypeAttr = "dagger.io/testctx.type"
-const TestCtxNameAttr = "dagger.io/testctx.name"
-const TestCtxPrewarmAttr = "dagger.io/testctx.prewarm"
```

Comment:
am i missing it somewhere or did we lose this metadata?

(i did read testctx and i can tell that at least `type` and `suite` are settable)